src/main/scala/edu/ie3/simona/model/thermal/DomesticHotWaterStorage.scala

```diff
@@ -0,0 +1,181 @@
+/*
+ * © 2024. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+
+package edu.ie3.simona.model.thermal
+
+import edu.ie3.datamodel.models.OperationTime
+import edu.ie3.datamodel.models.input.OperatorInput
+import edu.ie3.datamodel.models.input.thermal.{
+  DomesticHotWaterStorageInput,
+  ThermalBusInput,
+}
+import edu.ie3.simona.model.thermal.ThermalStorage.ThermalStorageState
+import edu.ie3.simona.model.thermal.ThermalStorage.ThermalStorageThreshold.{
+  StorageEmpty,
+  StorageFull,
+}
+import edu.ie3.util.quantities.PowerSystemUnits
+import edu.ie3.util.scala.quantities.DefaultQuantities.*
+import edu.ie3.util.scala.quantities.QuantityConversionUtils.toSquants
+import squants.energy.Kilowatts
+import squants.time.Seconds
+import squants.{Energy, Power}
+
+import java.util.UUID
+
+/** A domestic hot water storage used for implementations, which require a
+  * mutable storage. <p> <strong>Important:</strong> The field storageLvl is a
+  * variable.
+  *
+  * @param uuid
+  *   the element's uuid
+  * @param id
+  *   the element's human-readable id
+  * @param operatorInput
+  *   Operator input
+  * @param operationTime
+  *   Operation time
+  * @param bus
+  *   Thermal bus input
+  * @param maxEnergyThreshold
+  *   Maximum permissible energy stored in the storage
+  * @param pThermalMax
+  *   Thermal power, that can be charged / discharged
+  * @param storedEnergy
+  *   Energy stored in the thermal storage
+  */
+final case class DomesticHotWaterStorage(
```

Comment:
Maybe this can be abstracted, since there are many duplicate lines (cf. `CylindricalThermalStorage`).